resources/qml/WelcomePages/AddLocalPrinterScrollView.qml

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2019 Ultimaker B.V.
+// Cura is released under the terms of the LGPLv3 or higher.
+
+import QtQuick 2.10
+import QtQuick.Controls 2.3
+
+import UM 1.3 as UM
+import Cura 1.0 as Cura
+
+
+//
+// This is the scroll view widget for adding a (local) printer. This scroll view shows a list view with printers
+// categorized into 3 categories: "Ultimaker", "Custom", and "Other".
+//
+ScrollView
+{
+    id: base
+
+    property var currentItem: (machineList.currentIndex >= 0)
+                              ? machineList.model.getItem(machineList.currentIndex)
+                              : null
+    property string currentSection: preferredCategory
+    property string preferredCategory: "Ultimaker"
+
+    ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+    ScrollBar.vertical.policy: ScrollBar.AsNeeded
+
+    property int maxItemCountAtOnce: 10  // show at max 10 items at once, otherwise you need to scroll.
+    height: maxItemCountAtOnce * UM.Theme.getSize("action_button").height
+
+    clip: true
```

Comment:
Is the clipping really needed? Because if not, it's better not to have it on (because it's expensive)

Reply:
I tried without `clip: true`, and I got this... Maybe we can solve this differently.
![image](https://user-images.githubusercontent.com/2630468/54811895-10a5a600-4c8a-11e9-9fdd-637c75711f3e.png)

Reply:
Note that the section texts going above is a Qt bug that hasn't been resolved yet.

Reply:
In general I almost always set `clip: true` for scroll views because you usually want things to disappear when you scroll away from them. It's expensive but quite necessary for scrolling subviews.